lib/run_loop/ipa.rb

```diff
@@ -42,96 +35,59 @@ def inspect
     def bundle_identifier
-      if bundle_dir.nil? || !File.exist?(bundle_dir)
-        raise "Expected a '#{File.basename(path).split('.').first}.app'\nat path '#{payload_dir}'"
-      end
-
-      @bundle_identifier ||= lambda {
-        info_plist_path = File.join(bundle_dir, 'Info.plist')
-        unless File.exist? info_plist_path
-          raise "Expected an 'Info.plist' at '#{bundle_dir}'"
-        end
-        identifier = plist_buddy.plist_read('CFBundleIdentifier', info_plist_path)
-
-        unless identifier
-          raise "Expected key 'CFBundleIdentifier' in '#{info_plist_path}'"
-        end
-        identifier
-      }.call
+      app.bundle_identifier
     end
 
     # Inspects the app's Info.plist for the executable name.
     # @return [String] The value of CFBundleExecutable.
-    # @raise [RuntimeError] If the plist cannot be read or the
-    #   CFBundleExecutable is empty or does not exist.
     def executable_name
-      if bundle_dir.nil? || !File.exist?(bundle_dir)
-        raise "Expected a '#{File.basename(path).split('.').first}.app'\nat path '#{payload_dir}'"
-      end
-
-      @executable_name ||= lambda {
-        info_plist_path = File.join(bundle_dir, 'Info.plist')
-        unless File.exist? info_plist_path
-          raise "Expected an 'Info.plist' at '#{bundle_dir}'"
-        end
-        name = plist_buddy.plist_read('CFBundleExecutable', info_plist_path)
-
-        unless name
-          raise "Expected key 'CFBundleExecutable' in '#{info_plist_path}'"
-        end
-        name
-      }.call
+      app.executable_name
     end
 
-    # Inspects the app's file for the server version
+    # Inspects the app's executables for the server version
+    # @return[RunLoop::Version] a version instance
     def calabash_server_version
-      if bundle_dir.nil? || !File.exist?(bundle_dir)
-        raise "Expected a '#{File.basename(path).split('.').first}.app'\nat path '#{payload_dir}'"
-      else
-        if !executable_name.nil? && executable_name != ''
-          path_to_bin = File.join(bundle_dir, executable_name)
-          xcrun ||= RunLoop::Xcrun.new
-          hash = xcrun.exec(["strings", path_to_bin])
-          unless hash.nil?
-            version_str = hash[:out][/CALABASH VERSION: \d+\.\d+\.\d+/, 0]
-            unless version_str.nil? || version_str == ""
-              server_ver = version_str.split(":")[1].delete(' ')
-              RunLoop::Version.new(server_ver)
-            end
-          end
-        end
-      end
+      app.calabash_server_version
     end
 
     private
 
+    # @!visibility private
     def tmpdir
       @tmpdir ||= Dir.mktmpdir
     end
 
+    # @!visibility private
     def payload_dir
-      @payload_dir ||= lambda {
+      @payload_dir ||= lambda do
         FileUtils.cp(path, tmpdir)
         zip_path = File.join(tmpdir, File.basename(path))
         Dir.chdir(tmpdir) do
           system('unzip', *['-q', zip_path])
```

Comment:
E.g., for resigning we do:
```golang
        // NOTE: To correctly preserve MACOS metadata we must use ditto instead of zip.
	// Failing to do so can make codesign bail out with some very exotic
	// error messages in some rare cases.
	ShellOrDie("ditto", "-x", "-k", "--sequesterRsrc", filepath, tmpDir)
```